app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -100,4 +100,52 @@ public function isProductReviewInviteEnabled(): bool {
         return true;
     }
 
+    public function hasCorrectCredentials(?string $webshop_id, ?string $api_key): bool {
+        if (!trim($webshop_id) || !trim($api_key)) {
+            return false;
+        }
+        if ($webshop_id != $this->getShopId() || !hash_equals($api_key, $this->getApiKey())) {
+            return false;
+        }
+        return true;
+    }
+
+    public function syncProductReview(array $product_review): array {
+        $connection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        try {
+            $connection->beginTransaction();
+
+            ($review = Mage::getModel('review/review'))
+                ->setEntityPkValue($product_review['product_review']['product_id'])
+                ->setStatusId(Mage_Review_Model_Review::STATUS_PENDING)
+                ->setTitle($product_review['product_review']['title'])
+                ->setDetail($product_review['product_review']['review'])
+                ->setEntityId($review->getEntityIdByCode(Mage_Review_Model_Review::ENTITY_PRODUCT_CODE))
+                ->setStoreId(Mage::app()->getStore()->getStoreId())
+                ->setStores([Mage::app()->getStore()->getStoreId()])
+                ->setCustomerId($this->getCustomerId($product_review['product_review']['reviewer']['email']))
+                ->setNickname($product_review['product_review']['reviewer']['name'])
+                ->save();
+
+            Mage::getModel('rating/rating')
+                ->setRatingId(Mage::getStoreConfig('webwinkelconnect/product_review_invites/rating'))
```

Comment:
Should we check in `isProductReviewInviteEnabled` that this configuration option is set, or is that always the case or will it not cause problems if it isn't?

Reply:
I've set it up in such a way, that the dropdown always has one rating option selected. This means that if a user goes to the config page and saves it, even without choosing a rating option, the first one is saved in the db. I will check if there are any rating options created in the first place and warn the user to create them if there aren't any.